redaxo/src/core/lib/Fragment/Component/Choice.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Redaxo\Core\Fragment\Component;
+
+use Closure;
+use Redaxo\Core\Fragment\Slot;
+use rex_fragment;
+
+use function call_user_func;
+use function is_array;
+use function is_callable;
+
+class Choice extends rex_fragment
+{
+    /** @var array<string, array<string, string>|string> */
+    private array $choicesByLabels = [];
+
+    /** @var array<string, string> */
+    private array $choicesByValues = [];
+
+    public function __construct(
+        /**
+         * The choice label. Alternatively, you can use the
+         * label attribute.
+         */
+        public ?Slot $slotLabel = null,
+
+        /**
+         * Text that describes how to use the choice.
+         * Alternatively, you can use the notice property.
+         */
+        public ?Slot $slotNotice = null,
+
+        /**
+         * The choice label. If you need to display HTML,
+         * use the label slot instead.
+         */
+        public ?string $label = null,
+
+        /**
+         * The name of the choice.
+         */
+        public ?string $name = null,
+
+        /**
+         * The current values of the choice.
+         *
+         * @var null|string|array<string>
+         */
+        public null|string|array $value = null,
+
+        /**
+         * Choices is an array, where the array key is the
+         * item's label and the array value is the item's
+         * value.
+         *
+         * @var array<string, array<string, string>|string>
+         */
+        public array $choices = [],
+
+        /**
+         * The choice notice. If you need to display HTML,
+         * use the notice slot instead.
+         */
+        public ?string $notice = null,
+
+        /**
+         * Disables the choice.
+         */
+        public bool $disabled = false,
+
+        /**
+         * Allows more than one option to be selected.
+         */
+        public bool $multiple = false,
+
+        /**
+         * Placeholder text to show as a hint when the choice
+         * is empty.
+         */
+        public ?string $placeholder = null,
+
+        /**
+         * Makes the choice a required field.
+         */
+        public bool $required = false,
+
+        /**
+         * The choice type.
+         */
+        public ChoiceType $type = ChoiceType::Select,
+
+        /** @var array<string, string|int> */
+        public array $attributes = [],
+
+        /**
+         * By default, the array key of each item in the
+         * choices option is used as the text that's shown
+         * to the user. The choiceLabel option allows you
+         * to take more control.
+         */
+        public null|string|Closure $choiceLabel = null,
+    ) {
+        foreach ($this->choices as $choiceLabel => $choiceValue) {
```

Comment:
Ich denke, die Logik kommt hier im Constructor zu früh. Das sind oben ja alles Properties, die Optional nach dem Constructor auch noch separat gesetzt werden können.
Die Logik muss daher eher in render(), denke ich.

Reply:
siehe 6be25f19a08a222884ac596d48686fb7b2f4538d